sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -413,27 +442,48 @@ object RmmRapidsRetryIterator extends Arm {
         doSplit = false
         try {
           // call the user's function
+          if (config.testRetryOOMInjectionEnabled && !injectedOOM) {
+            injectedOOM = true
+            // ensure we have associated our thread with the running task, as
+            // `forceRetryOOM` requires a prior association.
+            RmmSpark.associateCurrentThreadWithTask(TaskContext.get().taskAttemptId())
+            RmmSpark.forceRetryOOM(RmmSpark.getCurrentThreadId)
+          }
           result = Some(attemptIter.next())
+          clearInjectedOOMIfNeeded()
         } catch {
-          case retryOOM: RetryOOM =>
-            if (lastException != null) {
-              retryOOM.addSuppressed(lastException)
-            }
-            lastException = retryOOM
-          case splitAndRetryOOM: SplitAndRetryOOM => // we are the only thread
-            if (lastException != null) {
-              splitAndRetryOOM.addSuppressed(lastException)
+          case ex: Throwable =>
```

Comment:
```suggestion
          case NonFatal(ex) =>
```

Reply:
So this is doing some handle for all types of exception that I think are worthwhile, like clearing any injected OOM and also wrapping a prior Retry/SplitAndRetryOOM as a likely cause of the eventual fatal exception. As such I think this might be a place where it's ok to go wide...

Reply:
The point of GpuOOM is that you should not "go wide"

Reply:
True, but this is a retry loop. The first time around we would have gone into this code and attempted an operation, which resulted in a GpuOOM, at which point we will retry the operation (the first time around). Assume the second time we get an OutOfMemoryError from the java side, the code will take the OutOfMemoryError as the top exception, add the prior GpuOOM to its caused-by and then throw the OutOfMemoryError. This is what I meant, this case works for either fatal or non-fatal exceptions and is doing something useful so I didn't see the need to split this out into a different `case` where I would want to add similar logic.